src/main/scala/com/fulcrumgenomics/umi/SimpleConsensusCaller.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.FgBioDef.forloop
+import com.fulcrumgenomics.util.NumericTypes.LogProbability
+
+/**
+  * A trait that can be mixed in to call sequences represented as strings of the same length.
+  *
+  * If a non-ACGTN bases is encountered at any position in the strings, that position is not consensus called, and the
+  * character from the first sequenced is used.
+  *
+  * @param errorRatePreLabeling The error probability prior to adding the UMI.  By default, set to almost surely zero.
+  * @param errorRatePostLabeling The error probability post to adding the UMI.  By default, set to almost surely zero
+  * @param pError The adjusted error probability.  By default set to ln(0.99), as the actual values don't matter much.
+  * @param pTruth The adjusted truth probability.  By default set to ln(0.01), as the actual values don't matter much.
+  */
+private[umi] class SimpleConsensusCaller(val errorRatePreLabeling: Byte = 90.toByte,
+                                         val errorRatePostLabeling: Byte = 90.toByte,
+                                         val pError: LogProbability = -4.60517,
```

Comment:
It might be nice to re-express these as Q scores here and transform them internally to be more consistent?